### PR TITLE
New memory defaults

### LINKED ISSUE #1904: cli: default --cache and --max-sql-memory to 128MiB
PR: https://github.com/cockroachdb/cockroach/pull/18040

### PR DESCRIPTION
**Pull request by jseldess**
- Update `cockroach start` docs
- Update cache and sql memory section of recommended prod settings
- Remove rec to reduce memory from local deployment tutorials
- Update prod deployment tutorials with new memory settings
- Update FAQ about memory usage

Fixes #1904

### CODE REVIEW

File: v1.1/deploy-cockroachdb-on-aws.md [hunk 1]
```diff
@@ -257,15 +257,22 @@ Locally, you'll need to [create the following certificates and keys](create-secu
...
+    $ cockroach start \
     --certs-dir=certs \
-    --advertise-host=<node1 internal IP address>
+    --advertise-host=<node1 internal IP address> \
+    --cache=25% \
+    --max-sql-memory=25% \
+    --background
     ~~~
 
+    This command specifies the location of certificates and the address at which other nodes can reach it. It also increases the node's cache and temporary SQL memory size to 25% of available system memory in order to improve read performance and prevent out-of-memory errors.
```
Comment (petermattis) [inline v1.1/deploy-cockroachdb-on-aws.md]: Perhaps `prevent out-of-memory errors when performing large queries`. @knz?

File: v1.1/recommended-production-settings.md [hunk 2]
```diff
@@ -44,15 +44,15 @@ For details about controlling the number and location of replicas, see [Configur
 
 {% include faq/clock-synchronization.html %}
 
-## Cache Size
+## Cache and SQL Memory Size <span class="version-tag">Changed in v1.1</span>
 
-If you run multiple applications on the same machine as a CockroachDB node, you might consider manually setting the cache size instead of using the default 25% of available memory.
+By default, each node's cache and temporary SQL memory size is `128MiB` respectively. This default is optimal for running a local development cluster with multiple nodes on a single host. When running a production cluster with one node per host, however, these defaults may lead to out-of-memory errors. To prevent such errors and improve read performance, it's recommended to manually increase the cache and SQL memory size to 25% or more of available system memory.
```
Comment (petermattis) [inline v1.1/recommended-production-settings.md]: I shy away from using the term `optimal`. Perhaps `This default is reasonable for running...`.

I think it is worthwhile to distinguish the effects of the flags. `--cache` will improve read performance. `--max-sql-memory` will increase the number of concurrent SQL queries that can be processed and the complexity of the queries that can be processed.

Cc @knz

**Comment by knz** [comment]
increasing --max-sql-size can be done for any combination of the following two reasons:
- increase the number of simultaneous client connections per node (the default of 128MB supports up to 6200 connections max)
- increase the capacity for in-memory processing of rows when using ORDER BY, GROUP BY, DISTINCT, joins and window functions.

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, 10 of 10 files at r4.
Review status: 15 of 16 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[v1.1/deploy-cockroachdb-on-aws-insecure.md, line 131 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-Ktw-W597MgggGlLuP6o:-Ktw-W597MgggGlLuP6p:byso0c0) ([raw file](https://github.com/cockroachdb/docs/blob/2129d1383b9c53f051fc14a5b15156e8bd3d6bf1/v1.1/deploy-cockroachdb-on-aws-insecure.md#L131)):*
> ```Markdown
> 	~~~
> 
> 	This command sets the node to insecure and identifies the address at which other nodes can reach it. It also increases the node's cache and temporary SQL memory size to 25% of available system memory in order to improve read performance and prevent out-of-memory errors.
> ```

I'm not sure the phrasing "sets a node to insecure" is accurate or informative. I would suggest "This commands starts an insecure node and identifies ..."

---

*[v1.1/deploy-cockroachdb-on-aws.md, line 272 at r5](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-KtrnRpSOdEiSjC9V0RA-r5-272:-Ktw115rJ3cPcdxMzTlt:b-ava3bi) ([raw file](https://github.com/cockroachdb/docs/blob/9fa4b80248cbb84e6dd4ec63f4c3a858f564f396/v1.1/deploy-cockroachdb-on-aws.md#L272)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Perhaps `prevent out-of-memory errors when performing large queries`. @knz?
</blockquote></details>

See my note above.

Also could you find a way to share this text across the different deployment docs? The current redundancy makes reviewing difficult.

---

*[v1.1/recommended-production-settings.md, line 49 at r5](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-KtrnRpL2gdNvYA6imw9-r5-49:-Ktw-8VG7Rl6SF56Ji98:b-dz2s29) ([raw file](https://github.com/cockroachdb/docs/blob/9fa4b80248cbb84e6dd4ec63f4c3a858f564f396/v1.1/recommended-production-settings.md#L49)):*
> This default is optimal for running a local development cluster with multiple nodes on a single host.

Replace this sentence by "This default was chosen to facilitate development and testing, where users are likely to run multiple CockroachDB nodes on a single computer."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-:-Ktw-rOLZ5mrxJsUyCni:b-h5m2xe)*

**Comment by knz** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1906)*

**Comment by jseldess** [comment]
Added these details to `recommended-production-settings.md`, @knz. Thank you.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[v1.1/deploy-cockroachdb-on-aws-insecure.md, line 131 at r4](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-Ktw-W597MgggGlLuP6o:-KtwHjXvWjYbSN9MS3h7:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/2129d1383b9c53f051fc14a5b15156e8bd3d6bf1/v1.1/deploy-cockroachdb-on-aws-insecure.md#L131)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I'm not sure the phrasing "sets a node to insecure" is accurate or informative. I would suggest "This commands starts an insecure node and identifies ..."
</blockquote></details>

Done.

---

*[v1.1/deploy-cockroachdb-on-aws.md, line 272 at r5](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-KtrnRpSOdEiSjC9V0RA-r5-272:-KtwHrqJLtwA_7kqXtkn:baamfwb) ([raw file](https://github.com/cockroachdb/docs/blob/9fa4b80248cbb84e6dd4ec63f4c3a858f564f396/v1.1/deploy-cockroachdb-on-aws.md#L272)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

See my note above.

Also could you find a way to share this text across the different deployment docs? The current redundancy makes reviewing difficult.
</blockquote></details>

Done. 

I know it's hard to review so much redundant content, and I appreciate your patience and tolerance here. We have an issue in the backlog to consolidate with includes: https://github.com/cockroachdb/docs/issues/1718

---

*[v1.1/recommended-production-settings.md, line 49 at r5](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-KtrnRpL2gdNvYA6imw9-r5-49:-KtwI8L_74l7t8Vz-SMd:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9fa4b80248cbb84e6dd4ec63f4c3a858f564f396/v1.1/recommended-production-settings.md#L49)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

> This default is optimal for running a local development cluster with multiple nodes on a single host.

Replace this sentence by "This default was chosen to facilitate development and testing, where users are likely to run multiple CockroachDB nodes on a single computer."
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-:-KtwIBTKTUtTCzDHPyWK:br29zdz)*

**Comment by jseldess** [comment]
Btw, I spoke to @a-robinson about required changes to k8 and docker swarm. He's going to open a separate PR updating the docker swarm docs (with new memory defaults as well as the `--init` command), and he's going to update the k8 configuration in conjunction with the actual 1.1 release.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-:-KtwIT2mTS8H7JbeMaas:b-tcqzib)*

**Comment by knz** [comment]
:lgtm:

---

Reviewed 14 of 14 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-:-Ktwfd9BGrCdAP6kw1O8:bnfp4nl)*

**Comment by jseldess** [comment]
Merging so I can start work on the `--init` command. @petermattis, if you have further feedback, I can integrate changes in a follow-up PR.

**Comment by petermattis** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[v1.1/deploy-cockroachdb-on-aws.md, line 272 at r5](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-KtrnRpSOdEiSjC9V0RA-r5-272:-Ku-i20vySBHCSdarYP2:b75n4lh) ([raw file](https://github.com/cockroachdb/docs/blob/9fa4b80248cbb84e6dd4ec63f4c3a858f564f396/v1.1/deploy-cockroachdb-on-aws.md#L272)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

Done. 

I know it's hard to review so much redundant content, and I appreciate your patience and tolerance here. We have an issue in the backlog to consolidate with includes: https://github.com/cockroachdb/docs/issues/1718
</blockquote></details>

The new wording SGTM.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1906#-:-Ku-i5rxhSH9xXyjWyN8:bnfp4nl)*

**Pull request merged**